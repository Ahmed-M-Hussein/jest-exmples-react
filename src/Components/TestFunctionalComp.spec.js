import React from "react";
import { create, act } from "react-test-renderer";
import { Button, Text } from "react-native";
import DemoFunctionalComponent from "../Demos/DemoFunctionalComponent";

describe("Test Class Component", () => {
  it("renders name Ahmed correctly", () => {
    const tree = create(<DemoFunctionalComponent name="Ahmed" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("test  inside DemoComponent", () => {
    let wrapper;
    act(() => {
      wrapper = create(<DemoFunctionalComponent />);
    });

    // expect(wrapper).toMatchSnapshot();

    act(() => {
      wrapper.update(<DemoFunctionalComponent name="Ahmed" />);
    });

    // expect(wrapper).toMatchSnapshot();

    act(() => {
      wrapper.update(<DemoFunctionalComponent name="Mohamed" />);
    });

    expect(wrapper).toMatchSnapshot();
  });
  it("call onPress correctly", () => {
    let wrapper;
    act(() => {
      wrapper = create(<DemoFunctionalComponent name="Ahmed" />);
    });
    const instance = wrapper.root;
    const button = instance.findByType(Button);
    const texts = instance.findAllByType(Text);
    act(() => button.props.onPress());
    act(() => button.props.onPress());
    expect(texts[1].props.children[1]).toBe(2);
  });

  it("unmount correctly", () => {
    const onUnmount = jest.fn();
    let wrapper;
    act(() => {
      wrapper = create(
        <DemoFunctionalComponent name="Ahmed" onUnmount={onUnmount} />
      );
    });

    act(() => {
      wrapper.unmount();
    });
    expect(onUnmount).toHaveBeenCalled();
  });
});
