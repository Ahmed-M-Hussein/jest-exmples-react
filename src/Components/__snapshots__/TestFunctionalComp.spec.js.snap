// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Class Component renders name Ahmed correctly 1`] = `
<View>
  <Text>
     Welcome 
    Ahmed
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Class Component test  inside DemoComponent 1`] = `
<View>
  <Text>
     Welcome 
    Guest
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Class Component test  inside DemoComponent 2`] = `
<View>
  <Text>
     Welcome 
    Ahmed
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Class Component test  inside DemoComponent 3`] = `
<View>
  <Text>
     
    Ahmed
     has logged out 
  </Text>
  <Text>
     Welcome 
    Mohamed
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;
